Jenkinsfile

```diff
@@ -36,35 +37,82 @@ pipeline {
                             steps {
                                 sh 'docker version'
                                 sh 'docker info'
+                                sh '''
+                                echo "check-config.sh version: ${CHECK_CONFIG_COMMIT}"
+                                curl -fsSL -o ${WORKSPACE}/check-config.sh "https://raw.githubusercontent.com/moby/moby/${CHECK_CONFIG_COMMIT}/contrib/check-config.sh" \
+                                && bash ${WORKSPACE}/check-config.sh || true
+                                '''
                             }
                         }
                         stage("Build dev image") {
+                            steps {
+                                sh 'docker build --force-rm --build-arg APT_MIRROR -t docker:${GIT_COMMIT} .'
+                            }
+                        }
+                        stage("Validate") {
                             steps {
                                 sh '''
-                                # todo: include ip_vs in base image
-                                sudo modprobe ip_vs
-                
-                                GITCOMMIT=$(git rev-parse --short HEAD)
-                                docker build --force-rm --build-arg APT_MIRROR -t docker:$GITCOMMIT .
+                                docker run --rm -t --privileged \
+                                  -v "$WORKSPACE/bundles:/go/src/github.com/docker/docker/bundles" \
+                                  -v "$WORKSPACE/.git:/go/src/github.com/docker/docker/.git" \
+                                  --name docker-pr$BUILD_NUMBER \
+                                  -e DOCKER_GITCOMMIT=${GIT_COMMIT} \
+                                  -e DOCKER_GRAPHDRIVER \
+                                  docker:${GIT_COMMIT} \
+                                  hack/validate/default
                                 '''
                             }
                         }
-                        stage("Run tests") {
+                        stage("Docker-py") {
```

Comment:
I was a bit cautious with the order of these test, because they all bind-mount the `bundles` directory, and some of the tests will clean that up, which makes archiving artefacts from previous stages complicated (not sure if there's a recommended approach for that?)